### ISSUE TITLE
Segment count value in repair status not correct for subrange repair

### ISSUE THREAD

**Issue by milannister** [thelastpickle/cassandra-reaper#61]
When repair is started with "Incremental repair : false" the value displayed on UI (and the value stored in back end storage) denotes number of nodes in the cluster instead the number defined in configuration with `segmentCount` property.

It appears that the problem is in CommonTools class at line:

```java
    RepairRun repairRun = storeNewRepairRun(context, cluster, repairUnit, cause, owner, nodes.keySet().size(), repairParallelism, intensity);
```

where `nodes.keySet().size()` is passed as a value of `int segments` parameter. I assume, in case when `incrementarRepair` is `false`, that `tokenSegments.size()` should be passed instead.

If this is, indeed, a bug, I can fix it. I would like to start contributing to the cassandra-reaper.

Also, what is the role of `segmentCount` property when incremental repairs are run if that number is just number of nodes? Or I am missing something?

**Comment by adejanovski** [comment]
HI,

this is deliberate actually.

Incremental repair doesn't handle concurrency very well, and it definitely does not work with subrange repair (anticompaction is disactivated in Cassandra with subrange repair).

And if you try to run it on more than one node at a time, you'll get in Validation Error hell, because an SSTable cannot be validation compacted and anticompacted at the same time.

I forced Reaper to create a single "fake" segment per node, so that we only process one node at a time.

If you want to perform subrange repair just set incremental to false (which you should do unless you're using TWCS, which might be the only compaction strategy that could work correctly with incremental repair).

**Comment by milannister** [comment]
Yes, but `nodes.keySet().size()` can be passed only in case when incremental repair is true. If incremental repair is `false`, that means that subrange repair is run and `tokenSegments.size()` should be passed as argument, right?

**Comment by adejanovski** [comment]
Hi @milannister ,

the number of segments will get overriden in `storeNewRepairSegment()` : 
```
private static void storeNewRepairSegments(AppContext context, List<RingRange> tokenSegments,
                                             RepairRun repairRun, RepairUnit repairUnit) {
    List<RepairSegment.Builder> repairSegmentBuilders = Lists.newArrayList();
    for (RingRange range : tokenSegments) {
      RepairSegment.Builder repairSegment = new RepairSegment.Builder(repairRun.getId(), range,
                                                                      repairUnit.getId());
      repairSegmentBuilders.add(repairSegment);
    }
    context.storage.addRepairSegments(repairSegmentBuilders, repairRun.getId());
    if (repairRun.getSegmentCount() != tokenSegments.size()) {
      LOG.debug("created segment amount differs from expected default {} != {}",
                repairRun.getSegmentCount(), tokenSegments.size());
      context.storage.updateRepairRun(
          repairRun.with().segmentCount(tokenSegments.size()).build(repairRun.getId()));
    }
  }
```

Here you can see that we call `udpateRepairRun` and feed it with `tokenSegments.size()`.

**Comment by milannister** [comment]
Hi @adejanovski 

You are right. It appears that the value is overwritten.

However, based on the schedule:
![reaper-schedule](https://cloud.githubusercontent.com/assets/4706789/23852000/13ceb2bc-07e6-11e7-94ff-22d8045be24f.png)

I would expect that segment count on repair result screen is not equal to the number of nodes (which is 20 in our case), but to the value on schedule screen:
![reaper-segment-count](https://cloud.githubusercontent.com/assets/4706789/23852001/15a13ccc-07e6-11e7-978a-6a7cc8f01ec8.png)

Why is this happening?

**Comment by adejanovski** [comment]
Hi @milannister,

I can reproduce the bug locally and will issue a fix ASAP.

**Comment by adejanovski** [comment]
@milannister, 

fix has been merged into master.
The display wasn't right but the number of segments was correct.

Please confirm that the fix works for you.

Thanks

**Comment by adejanovski** [comment]
Fix released in 0.5.0

**Comment by milannister** [comment]
Sorry for late response. We are currently testing the cluster without repair. Will confirm the fix locally as soon as I setup the environment.
Thanks for being so responsive!
